File: recursion/towerOfHanoi/towerOfHanoi.js
function towersOfHanoi(n, fromPeg, toPeg, auxPeg) {
    if (n === 1) {
        console.log(`Move disk 1 from peg ${fromPeg} to peg ${toPeg}`);
        return;
    }
    towersOfHanoi(n - 1, fromPeg, auxPeg, toPeg);
    console.log(`Move disk ${n} from peg ${fromPeg} to peg ${toPeg}`);
    towersOfHanoi(n - 1, auxPeg, toPeg, fromPeg);
}

// Example usage
towersOfHanoi(3, 'main', 'destiny', 'aux');

File: recursion/towerOfHanoi/towerOfHanoi.c
#include <stdio.h>

void TowersOfHanoi(int n, char frompeg, char topeg, char auxpeg) {
    if (n == 1) {
        printf("Move disk 1 from peg %c to peg %c\n", frompeg, topeg);
        return;
    }
    TowersOfHanoi(n - 1, frompeg, auxpeg, topeg);
    printf("Move disk %d from peg %c to peg %c\n", n, frompeg, topeg);
    TowersOfHanoi(n - 1, auxpeg, topeg, frompeg);
}

int main() {
    int n = 3; // You can change this to the desired number of disks
    TowersOfHanoi(n, 'A', 'C', 'B');
    return 0;
}
